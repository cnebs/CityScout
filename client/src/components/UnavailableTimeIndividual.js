import React from 'react';
import { ListItem, ListItemText } from '@material-ui/core';
import moment from 'moment';
import DeleteRoundedIcon from '@material-ui/icons/DeleteRounded';

export default function UnavailableTimeIndividual({ time, handleDelete }) {
  const dayOfTheWeek = moment(time.time_start).format('dddd');
  const timeStart = moment(time.time_start).format('hh:mm a');
  const timeEnd = moment(time.time_end).format('hh:mm a');
  return (
    <ListItem>
      <ListItemText
        primary={time.name}
        secondary={`${dayOfTheWeek}s: ${timeStart} - ${timeEnd}`}
      ></ListItemText>
      <DeleteRoundedIcon onClick={() => handleDelete(time.item_id)} />
    </ListItem>
  );
}
