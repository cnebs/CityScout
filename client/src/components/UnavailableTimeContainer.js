import React from 'react';
import { List } from '@material-ui/core';
import UnavailableTimeIndividual from './UnavailableTimeIndividual';

export default function UnavailableTimeContainer({ times, handleDelete }) {
  console.log(times);
  return (
    <List>
      {times.map(time => (
        <UnavailableTimeIndividual time={time} handleDelete={handleDelete} />
      ))}
    </List>
  );
}
