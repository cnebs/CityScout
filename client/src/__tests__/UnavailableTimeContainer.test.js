import { shallow } from 'enzyme';
import React from 'react';
import UnavailableTimeContainer from '../components/UnavailableTimeContainer';
import UnavailableTimeIndividual from '../components/UnavailableTimeIndividual';
import { List } from '@material-ui/core/';

const handleDelete = jest.fn();

describe('Settings View component', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <UnavailableTimeContainer
        times={[{ foo: 'bar', id: 1 }, { baz: 'bo', id: 2 }]}
        handleDelete={handleDelete}
      />
    );
  });
  test('it renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('it renders the corect child components', () => {
    expect(wrapper.find(UnavailableTimeIndividual).length).toBe(2);
    expect(wrapper.find(List).length).toBe(1);
  });
});
