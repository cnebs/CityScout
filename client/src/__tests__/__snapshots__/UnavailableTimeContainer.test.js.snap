// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unavailable Time Container component it renders without crashing 1`] = `ShallowWrapper {}`;
